### ISSUE TITLE
OpenMPI header mpif.h fails to parse

### ISSUE THREAD

**Issue by mikebentley15** [NCAR/KGen#57]
The mpif.h file that comes with openmpi causes KGen to fail to parse.  It turns out it is because MPI_COMM_WORLD is not directly defined in that file, but rather in mpif-common.h which is included by mpif.h.  I believe the correct solution would be to resolve all inclusions before trying to search for specific strings within.

**Comment by mikebentley15** [comment]
I looked through the fortran parsing code and found out that includes are being processed correctly.  However, for some unknown reason (at least unknown to me :confused: ) the mpif.h file from openmpi is only finding two statements:

```
Specification_Part(
  Parameter_Stmt('PARAMETER',
    Named_Constant_Def(Name('MPI_MODE_CREATE'), Int_Literal_Constant('1', None))),
  Parameter_Stmt('PARAMETER',
    Named_Constant_Def(Name('MPI_MODE_EXCL'), Int_Literal_Constant('64', None)))
  )
```

**Comment by grnydawn** [comment]
Hi Michael,

Thanks for raising an issue about mpif.h header file. Also I appreciate for
your pull request which is quite reasonable and is applied to devel branch
now.
KGen is under active development these days. If you can share any
experience on using or developing KGen, please share it with me directly
through either "kgen@ucar.edu" or "youngsun@ucar.edu".

Thanks,

On Thu, May 26, 2016 at 9:08 PM, Michael Bentley notifications@github.com
wrote:

> I looked through the fortran parsing code and found out that includes are
> being processed correctly. However, for some unknown reason (at least
> unknown to me 😄 ) the mpif.h file from openmpi is only finding two
> statements:
> 
> Specification_Part(
>   Parameter_Stmt('PARAMETER',
>     Named_Constant_Def(Name('MPI_MODE_CREATE'), Int_Literal_Constant('1', None))),
>   Parameter_Stmt('PARAMETER',
>     Named_Constant_Def(Name('MPI_MODE_EXCL'), Int_Literal_Constant('64', None)))
>   )
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> https://github.com/NCAR/KGen/issues/57#issuecomment-222050045, or mute
> the thread
> https://github.com/notifications/unsubscribe/AKkETy6PmCxYLVwo2pEvy4C3kKu2JnPgks5qFmBBgaJpZM4IoCzT
> .

## 

Youngsung Kim

**Comment by grnydawn** [comment]
This issue is resolved by e08ae0826d237a48795cb1c770d91b3037d3d94e.
